mvcc/kvstore.go

```diff
@@ -160,6 +166,56 @@ func (s *store) Hash() (hash uint32, revision int64, err error) {
 	return h, s.currentRev, err
 }
 
+func (s *store) HashByRev(rev int64) (hash uint32, currentRev int64, compactRev int64, err error) {
+	readTx := s.b.ReadTx()
+	for {
```

Comment:
there's no need to wait on compaction with txnread; I should have made that more clear...